markdown.go

```diff
@@ -0,0 +1,297 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/juju/gnuflag"
+	"io"
+	"sort"
+	"strings"
+)
+
+// InfoCommand is a subset of Command methods needed to print the Markdown
+// document. In particular, all these methods are "static", hence should not
+// do anything scary or destructive.
+type InfoCommand interface {
+	// Info returns information about the Command.
+	Info() *Info
+	// SetFlags adds command specific flags to the flag set.
+	SetFlags(f *gnuflag.FlagSet)
+}
+
+// MarkdownOptions configures
+type MarkdownOptions struct {
+	// Title defines the title to print at the top of the document. If this
+	// field is empty, no title will be printed.
+	Title string
+	// UsagePrefix will be printed before the command usage (for example, the
+	// name of the supercommand).
+	UsagePrefix string
+	// LinkForCommand maps each "peer command" name (e.g. see also commands) to
+	// the link target for that command (e.g. a section of the Markdown doc, or
+	// a webpage).
+	LinkForCommand func(string) string
+	// LinkForSubcommand maps each sub-command name to the link target for that
+	//command (e.g. a section of the Markdown doc, or a webpage).
+	LinkForSubcommand func(string) string
+}
+
+// PrintMarkdown prints Markdown documentation about the given command to the
+// given io.Writer. The MarkdownOptions can be provided to customise the
+// output.
+func PrintMarkdown(w io.Writer, cmd InfoCommand, opts MarkdownOptions) error {
+	// TODO: error handling ?
```

Comment:
Yeah, we need some. But it might be easier to use a `bytes.Buffer`, ignore the errors on that, then `io.Copy` the result to `w`. On the `io.Copy`, that is the only error we care about.